pkg/cmd/experimental/sync-groups/interfaces.go

```diff
@@ -0,0 +1,33 @@
+package group
+
+import (
+	"github.com/go-ldap/ldap"
+
+	userapi "github.com/openshift/origin/pkg/user/api"
+)
+
+// LDAPGroupLister lists the groups that need to be synced by a job. The groups are represented with
+// a unique identifier that, when injected into the specific filter for groups and conjoined with the
```

Comment:
Don't dictate how the filters are chained and how this value is injected.  I think we only need to pin: 
```
LDAPGroupLister lists the groups that need to be synced by a job.  The LDAPGroupLister needs to be paired with an LDAPGroupDataExtractor that understands the format of the groups returned.
```